gptme/tools/shell.py

```diff
@@ -243,15 +243,27 @@ def execute_shell(
     """Executes a shell command and returns the output."""
     shell = get_shell()
     assert not args
+    whitelist_commands = ["ls", "stat", "cd", "cat", "pwd", "echo"]
+    whitelisted = True 
 
     cmd = code.strip()
     if cmd.startswith("$ "):
         cmd = cmd[len("$ ") :]
 
-    print_preview(cmd, "bash")
-    if not confirm("Run command?"):
-        yield Message("system", "Command not run")
-        return
+    #NOTE: This does not handle control flow words like if, for, while.
+    regex = r"(?:^|[|&;]|\|\||&&|\n)\s*([^\s|&;]+)"
```

Comment:
I tested the regex extensively but I guess it's the most unpredictable bit here.

Reply:
[regex testing site](https://regex101.com/r/1e57VF/1)